### PR TITLE
Cilium operator service account is missing node read permissions

### PR DESCRIPTION
**Pull request by silenteh**
This PR:

- allows the `cilium-operator` service account to read the  `nodes` resource from the k8s api.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Review by pipo02mix — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.5.1] - 2022-10-25
```
Comment (fiunchinho) [inline CHANGELOG.md:10]: We have automation to take care of releasing. This PR should only contain the changes you want in the app , and a new entry in the changelog file under the `Unreleased` section. Ping me, I can show you how to use the automation!

**Review by fiunchinho — CHANGES_REQUESTED** [review]
The changelog should only contain the unreleased changes.

**Comment by fiunchinho** [comment]
I'm checking one of my workload clusters and this doesn't seem to be required. The operator is not complaining, and the `ClusterRole` already has permissions to `list` `Nodes`.

File: CHANGELOG.md (hunk 1 above)
Comment (silenteh) [reply CHANGELOG.md]: Even though this PR might not be necessary anymore, I have fixed this

**Comment by silenteh** [comment]
> I'm checking one of my workload clusters and this doesn't seem to be required. The operator is not complaining, and the `ClusterRole` already has permissions to `list` `Nodes`.

As per our Slack discussion this seems to be affecting version 0.4.2 of the Cilium App but not the latest version.
So this PR does not seem to be needed anymore

**Pull request closed without merging**